### PR TITLE
CMake: ensure some more, modern style rules

### PR DESCRIPTION
**Pull request by tamiko**

### CODE REVIEW

File: cmake/macros/macro_clear_feature.cmake [hunk 1]
```diff
@@ -25,10 +25,7 @@
 #
 
 macro(CLEAR_FEATURE _feature)
```
Comment (tjhei) [inline cmake/macros/macro_clear_feature.cmake:27]:
```suggestion
macro(clear_feature _feature)
```?

File: cmake/setup_compiler_flags.cmake [hunk 2]
```diff
@@ -141,4 +141,4 @@ else(DEAL_II_SETUP_DEFAULT_COMPILER_FLAGS)
     "Skipped setup of default compiler flags "
     "(DEAL_II_SETUP_DEFAULT_COMPILER_FLAGS=OFF)"
     )
-endif(DEAL_II_SETUP_DEFAULT_COMPILER_FLAGS)
```
Comment (tjhei) [inline cmake/setup_compiler_flags.cmake:144]: beautiful!

**Review by tjhei — APPROVED** [review]

File: cmake/macros/macro_clear_feature.cmake (hunk 1 above)
Comment (tamiko) [reply cmake/macros/macro_clear_feature.cmake:27]: Let me lowercase all of the macro names in a separate PR.

Comment (tamiko) [reply cmake/macros/macro_clear_feature.cmake:27]: Actually. Let me add it to this PR.

**Comment by tamiko** [comment]
@tjhei Updated. Please re-review.

File: cmake/macros/macro_clear_feature.cmake [hunk 3]
```diff
@@ -24,11 +24,8 @@
 # DEAL_II_STRING_SUFFIXES to the corresponding DEAL_II_* variables
 #
 
-macro(CLEAR_FEATURE _feature)
-  foreach(_var ${DEAL_II_LIST_SUFFIXES})
-    unset(${_feature}_${_var})
-  endforeach()
-  foreach(_var ${DEAL_II_STRING_SUFFIXES})
+macro(clear_feature _feature)
+  foreach(_var ${DEAL_II_LIST_SUFFIXES} ${DEAL_II_STRING_SUFFIXES})
```
Comment (tamiko) [inline cmake/macros/macro_clear_feature.cmake:28]: I have smuggled in this small simplification as well.

Comment (tjhei) [reply cmake/macros/macro_clear_feature.cmake:28]: Saw that. :+1:

**Review by tjhei — APPROVED** [review]
Can you update the PR title? /rebuild

**Comment by tamiko** [comment]
@tjhei done.

**Review by drwells — APPROVED** [review]

**Comment by tamiko** [comment]
In reference to #12374

**Pull request merged**